master/master.cfg

```diff
@@ -1364,19 +1370,40 @@ def finish_dist(f, local_dist_dir, dist_subdirs, s3_addy, remote_dist_dir, compo
     f.addStep(MasterShellCommand(name="consolidate artifacts",
                                  command=["sh", "-c", WithProperties(consolidate_cmd)]))
 
-    # Build the channel manifest
+    # Build the channel manifest v1
     manifest_name = "channel-" + component + "-" + channel
     manifest_cmd = "ls " + final_dist_dir + " > " + manifest_name + " && mv " + manifest_name + " " + final_dist_dir + "/"
     f.addStep(MasterShellCommand(name="manifesting",
                                  command=["sh", "-c", WithProperties(manifest_cmd)]))
 
-    # Write cargo date
-    if "cargo" in component:
-        cargo_date_cmd = "date +\%Y-\%m-\%d > " + final_dist_dir + "/cargo-build-date.txt"
-        f.addStep(MasterShellCommand(name="Write date to cargo-build-date.txt",
-                                     command=["sh", "-c", cargo_date_cmd] ))
+    # Build the channel manifest v2, which is a toml file describing the components
+    # of Rust. This only works for the 'rust' component, which contains various
+    # artifacts of other builds. It depends on the v1 manifests and channel build
+    # date of the 'cargo' and 'rustc' channels.
+    if component == "rust":
+        manifest_v2_name = "channel-" + component + "-" + channel + ".toml"
+        manifest_v2_file = final_dist_dir + "/" + manifest_v2_name
+        # FIXME: This is a poor way to find the build-rust-manifest script
+        manifest_v2_cmd = ("python $HOME/rust-buildbot/master/build-rust-manifest.py " +
```

Comment:
Using globals will be unpleasant if we ever attempt to unit test the script. Based on what I've learned from attempting to do the same thing earlier, I think having the script separate may make it easier to add debugging information and tests in the future